src/main/java/com/gargoylesoftware/htmlunit/WebClient.java

```diff
@@ -1681,6 +1688,97 @@ else if (status == HttpStatus.SC_TEMPORARY_REDIRECT
         return webResponse;
     }
 
+    /**
+     * Validate the cached {@link WebResponse} to see if it could be used
+     * @param fromCache the {@link WebResponse} stored in cache that need to be validated before using
+     * @param webRequest the {@link WebRequest} that go with the {@link WebResponse} from cache
+     */
+    private WebResponse validateWebResponseFromCache(final WebResponse fromCache, final WebRequest webRequest) throws IOException {
```

Comment:
code below like `fromCache.getStatusCode()` are hard to read
```suggestion
    private WebResponse validateWebResponseFromCache(final WebResponse cached, final WebRequest webRequest) throws IOException {
```